### DISCUSSION TITLE
Issues with getting the name of clothing items using natives.

### DISCUSSION THREAD

**Discussion by aoqia194** [YimMenu/YimMenu#856] [Q&A] [answered]
I came back to coding and decided to fix my ped editor (it was really bad before)
As the title suggests, I am having issues with getting clothing item names.
[Screenshot of the UI with an example of my issue.](https://i.imgur.com/pvU1n2X.png)

I have wrote this code for getting the name:
```
ped_component component;
Hash component_hash = FILES::GET_HASH_NAME_FOR_COMPONENT(self::ped, i, k, 1 /* FIXME: 1 should not be used always */);

FILES::GET_SHOP_PED_COMPONENT(component_hash, reinterpret_cast<Any*>(&component));
const char* name = HUD::GET_FILENAME_FOR_AUDIO_CONVERSATION(component.m_textLabel);

ped_outfit_names[i].push_back(name);
```

I run this in a loop so `i` is the component_id and `k` is the drawable_id of the clothing.
Also, for some reason these values change a lot compared to some UI functions. For example, when you get a phone notification from Lester, some of the values will be NULL still but some will change to Lester (in this case in the screenshot it's `Crew Title Not Set`).
Very weird.

**Comment by pelecanidae** [comment, marked as answer]
Make sure to copy the string if you need a lifetime longer than the fiber pool scope. RAGE will free and later on reuse the same allocation for a different string, so you can't just try using the same pointer out of scope.

An easy way to do that is by using the implicit std::string conversion:

```
std::string name = HUD::GET_FILENAME_FOR_AUDIO_CONVERSATION(component.m_textLabel);
```

Also, you may want to cache these results if you aren't already. String allocations are cheap, but not when you're recycling thousands of them every game tick.

**Reply by aoqia194** [reply]
Good suggestion. I was wondering why some of the names were pointing to random clothing items that (im assuming) have been allocated to different text afterwards.
Also yes, I do cache them in a vector.

**Comment by Aure7138** [comment]
I think your problem is in communicating script thread with dx thread, you can find an example here https://github.com/YimMenu/YimMenu/blob/c138aaa9748d0f9c85f3c507bc2866c4b49bfb3a/src/views/vehicle/view_spawn_vehicle.cpp#L73-L79

**Reply by aoqia194** [reply]
I don't know what dx thread is but wouldn't that example be queued to the next game tick (like others have said) so veh_hash would always be 0? I am not quite taking away anything from that example. Could just be me though.

**Reply by aoqia194** [reply]
Edit: I will just make new discussion for different issue since this one has been solved!
